File: tuts/011-getting-started-batch-fargate/javascript/libs/batch-wrapper.js
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { BatchClient, CreateComputeEnvironmentCommand, CreateJobQueueCommand, DeleteComputeEnvironmentCommand, DeleteJobQueueCommand } from "@aws-sdk/client-batch";

export const createComputeEnv = async (client, params) => {
  const command = new CreateComputeEnvironmentCommand(params);
  return client.send(command);
};
export const createJobQueue = async (client, params) => {
  const command = new CreateJobQueueCommand(params);
  return client.send(command);
};
export const deleteJobQueue = async (client, params) => {
  const command = new DeleteJobQueueCommand(params);
  return client.send(command);
};
export const deleteComputeEnv = async (client, params) => {
  const command = new DeleteComputeEnvironmentCommand(params);
  return client.send(command);
};

File: tuts/011-getting-started-batch-fargate/python/batch_wrapper.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

import logging
import boto3
from botocore.exceptions import ClientError

logger = logging.getLogger(__name__)


class BatchWrapper:
    def __init__(self, client):
        self.client = client

    def create_compute_environment(self, name, subnets, security_groups, service_role):
        """Calls batch:CreateComputeEnvironment."""
        try:
            return self.client.create_compute_environment(computeEnvironmentName=name, type='MANAGED', computeResources={'type': 'FARGATE', 'maxvCpus': 4, 'subnets': subnets, 'securityGroupIds': security_groups}, serviceRole=service_role)['computeEnvironmentArn']
            logger.info("CreateComputeEnvironment succeeded.")
        except ClientError:
            logger.exception("CreateComputeEnvironment failed.")
            raise

    def create_job_queue(self, name, compute_env_arn):
        """Calls batch:CreateJobQueue."""
        try:
            return self.client.create_job_queue(jobQueueName=name, priority=1, computeEnvironmentOrder=[{'order': 1, 'computeEnvironment': compute_env_arn}])['jobQueueArn']
            logger.info("CreateJobQueue succeeded.")
        except ClientError:
            logger.exception("CreateJobQueue failed.")
            raise

    def delete_job_queue(self, name):
        """Calls batch:DeleteJobQueue."""
        try:
            self.client.delete_job_queue(jobQueue=name)
            logger.info("DeleteJobQueue succeeded.")
        except ClientError:
            logger.exception("DeleteJobQueue failed.")
            raise

    def delete_compute_environment(self, name):
        """Calls batch:DeleteComputeEnvironment."""
        try:
            self.client.delete_compute_environment(computeEnvironment=name)
            logger.info("DeleteComputeEnvironment succeeded.")
        except ClientError:
            logger.exception("DeleteComputeEnvironment failed.")
            raise
